provisioner/echo/serve.go

```diff
@@ -82,6 +82,10 @@ func (e *echo) Provision(stream proto.DRPCProvisioner_ProvisionStream) error {
 		return err
 	}
 	request := msg.GetStart()
+	if request == nil {
+		// A cancel could occur here!
+		return nil
```

Comment:
Is the cancellation inferable via context, and if so, should we consider doing something like this here?

```go
select {
case <-stream.Context().Done():
	return stream.Context().Err()
default
	return errors.New("")
}
```

Reply:
Nah, the cancelation isn't sent over the context at all. This should actually be fine because messages are handled synchronously, this one was just out of order!